File: AnalogClock.java
// AnalogClock.java - Аналоговые часы на Java (Swing)
import javax.swing.*;
import java.awt.*;
import java.awt.event.*;
import java.time.LocalTime;

public class AnalogClock extends JPanel implements ActionListener {
    private Timer timer;
    private int radius = 180;
    private Point center;
    private Color faceColor = new Color(245, 245, 220);
    private Color handColor = Color.DARK_GRAY;
    private Color secondColor = Color.RED;
    private boolean useRoman = false;

    public AnalogClock() {
        setPreferredSize(new Dimension(400, 400));
        setBackground(faceColor);
        timer = new Timer(1000, this);
        timer.start();
        addMouseListener(new MouseAdapter() {
            public void mouseClicked(MouseEvent e) {
                // переключение цвета при клике
                faceColor = (faceColor == Color.WHITE) ? new Color(245,245,220) : Color.WHITE;
                repaint();
            }
        });
        // JFrame
        JFrame frame = new JFrame("🕐 ClockForge - Java");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setResizable(false);
        frame.add(this);
        frame.pack();
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        Graphics2D g2 = (Graphics2D) g;
        g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
        center = new Point(getWidth()/2, getHeight()/2);
        int cx = center.x, cy = center.y;

        // Циферблат
        g2.setColor(faceColor);
        g2.fillOval(cx - radius, cy - radius, radius*2, radius*2);
        g2.setColor(Color.BLACK);
        g2.drawOval(cx - radius, cy - radius, radius*2, radius*2);

        // Метки
        LocalTime now = LocalTime.now();
        int hours = now.getHour() % 12;
        int minutes = now.getMinute();
        int seconds = now.getSecond();

        for (int i = 0; i < 12; i++) {
            double angle = Math.toRadians(i * 30 - 90);
            int x1 = cx + (int)((radius - 20) * Math.cos(angle));
            int y1 = cy + (int)((radius - 20) * Math.sin(angle));
            int x2 = cx + (int)((radius - 5) * Math.cos(angle));
            int y2 = cy + (int)((radius - 5) * Math.sin(angle));
            g2.drawLine(x1, y1, x2, y2);
            // Цифры
            String label;
            if (useRoman) {
                String[] roman = {"XII","I","II","III","IV","V","VI","VII","VIII","IX","X","XI"};
                label = roman[i];
            } else {
                label = (i+1) == 12 ? "12" : String.valueOf(i+1);
            }
            int tx = cx + (int)((radius - 40) * Math.cos(angle));
            int ty = cy + (int)((radius - 40) * Math.sin(angle));
            g2.setFont(new Font("Arial", Font.BOLD, 16));
            g2.setColor(Color.BLACK);
            g2.drawString(label, tx - 8, ty + 6);
        }

        // Стрелки
        drawHand(g2, Math.toRadians((hours + minutes/60.0) * 30 - 90), radius*0.5, handColor, 6);
        drawHand(g2, Math.toRadians((minutes + seconds/60.0) * 6 - 90), radius*0.7, handColor, 4);
        drawHand(g2, Math.toRadians(seconds * 6 - 90), radius*0.8, secondColor, 2);

        // Центр
        g2.setColor(Color.BLACK);
        g2.fillOval(cx - 8, cy - 8, 16, 16);
    }

    private void drawHand(Graphics2D g, double angle, double length, Color color, int width) {
        int cx = center.x, cy = center.y;
        int x = cx + (int)(length * Math.cos(angle));
        int y = cy + (int)(length * Math.sin(angle));
        g.setColor(color);
        g.setStroke(new BasicStroke(width, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        g.drawLine(cx, cy, x, y);
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        repaint();
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(AnalogClock::new);
    }
}

File: AnalogClock.cs
// AnalogClock.cs - Аналоговые часы на C# (WinForms)
using System;
using System.Drawing;
using System.Windows.Forms;

public class AnalogClock : Form
{
    private Timer timer;
    private int radius = 180;
    private Point center;
    private Color faceColor = Color.FromArgb(245,245,220);
    private Color handColor = Color.DarkGray;
    private Color secondColor = Color.Red;
    private bool useRoman = false;

    public AnalogClock()
    {
        Text = "🕐 ClockForge - C#";
        Size = new Size(450, 450);
        StartPosition = FormStartPosition.CenterScreen;
        DoubleBuffered = true;
        timer = new Timer();
        timer.Interval = 1000;
        timer.Tick += (s, e) => Invalidate();
        timer.Start();
        MouseClick += (s, e) => {
            faceColor = faceColor == Color.White ? Color.FromArgb(245,245,220) : Color.White;
            Invalidate();
        };
        // Кнопки управления
        Button bgBtn = new Button { Text = "Цвет фона", Location = new Point(20, 380), Size = new Size(100, 30) };
        bgBtn.Click += (s, e) => {
            ColorDialog cd = new ColorDialog();
            if (cd.ShowDialog() == DialogResult.OK) faceColor = cd.Color;
        };
        Controls.Add(bgBtn);
        Button handBtn = new Button { Text = "Цвет стрелок", Location = new Point(130, 380), Size = new Size(100, 30) };
        handBtn.Click += (s, e) => {
            ColorDialog cd = new ColorDialog();
            if (cd.ShowDialog() == DialogResult.OK) handColor = cd.Color;
        };
        Controls.Add(handBtn);
        Button romanBtn = new Button { Text = "Римские", Location = new Point(240, 380), Size = new Size(100, 30) };
        romanBtn.Click += (s, e) => { useRoman = !useRoman; Invalidate(); };
        Controls.Add(romanBtn);
    }

    protected override void OnPaint(PaintEventArgs e)
    {
        base.OnPaint(e);
        Graphics g = e.Graphics;
        g.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias;
        center = new Point(ClientSize.Width/2, ClientSize.Height/2 - 20);
        int cx = center.X, cy = center.Y;

        // Циферблат
        g.FillEllipse(new SolidBrush(faceColor), cx - radius, cy - radius, radius*2, radius*2);
        g.DrawEllipse(Pens.Black, cx - radius, cy - radius, radius*2, radius*2);

        DateTime now = DateTime.Now;
        int hours = now.Hour % 12;
        int minutes = now.Minute;
        int seconds = now.Second;

        for (int i = 0; i < 12; i++)
        {
            double angle = (i * 30 - 90) * Math.PI / 180;
            int x1 = cx + (int)((radius - 20) * Math.Cos(angle));
            int y1 = cy + (int)((radius - 20) * Math.Sin(angle));
            int x2 = cx + (int)((radius - 5) * Math.Cos(angle));
            int y2 = cy + (int)((radius - 5) * Math.Sin(angle));
            g.DrawLine(Pens.Black, x1, y1, x2, y2);
            string label;
            if (useRoman)
            {
                string[] roman = {"XII","I","II","III","IV","V","VI","VII","VIII","IX","X","XI"};
                label = roman[i];
            }
            else
            {
                label = (i+1) == 12 ? "12" : (i+1).ToString();
            }
            int tx = cx + (int)((radius - 40) * Math.Cos(angle));
            int ty = cy + (int)((radius - 40) * Math.Sin(angle));
            g.DrawString(label, new Font("Arial", 14, FontStyle.Bold), Brushes.Black, tx - 8, ty - 8);
        }

        // Стрелки
        DrawHand(g, (hours + minutes/60.0) * 30 - 90, radius*0.5, handColor, 6);
        DrawHand(g, (minutes + seconds/60.0) * 6 - 90, radius*0.7, handColor, 4);
        DrawHand(g, seconds * 6 - 90, radius*0.8, secondColor, 2);

        g.FillEllipse(Brushes.Black, cx - 8, cy - 8, 16, 16);
    }

    private void DrawHand(Graphics g, double angleDeg, double length, Color color, int width)
    {
        double angle = angleDeg * Math.PI / 180;
        int cx = center.X, cy = center.Y;
        int x = cx + (int)(length * Math.Cos(angle));
        int y = cy + (int)(length * Math.Sin(angle));
        using (Pen pen = new Pen(color, width)) { pen.StartCap = pen.EndCap = System.Drawing.Drawing2D.LineCap.Round;
            g.DrawLine(pen, cx, cy, x, y); }
    }

    [STAThread]
    static void Main() { Application.EnableVisualStyles(); Application.Run(new AnalogClock()); }
}
